.github/workflows/docker.yml

```diff
@@ -0,0 +1,80 @@
+name: (sub) Build Docker image
+
+defaults:
+  run:
+    shell: bash
+
+on:
+  workflow_call:
+    inputs:
+      publish:
+        description: A flag that triggers publishing to GHCR
+        type: boolean
+        default: false
+      python_version:
+        description: A version of Python to install
+        type: string
+        required: true
+
+jobs:
+
+  build:
+    name: Build ${{ inputs.publish && 'and publish' || '' }} a Docker image
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v3
+
+      - name: make sure we pull the latest commits
+        if: inputs.publish
+        run: git pull
+
+      - name: install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ inputs.python_version }}
+
+      - name: install Poetry
+        uses: Gr1N/setup-poetry@v8
+
+      - name: get the tag name for new image
+        id: tag
+        run: |
+          echo "version_tag=$(poetry version -s)" >> $GITHUB_OUTPUT
+
+      - name: build and prepare package for containerization
+        run: |
+          poetry env use ${{ inputs.python_version }}
+          poetry lock
+          poetry build
+          poetry export --without-hashes --format=requirements.txt > requirements.txt
+
+      - name: determine docker tags and labels
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ghcr.io/paloaltonetworks/panos_upgrade_assurance
+          tags: |
+            type=raw,value=latest
+            type=semver,pattern=v{{version}},value=${{ steps.tag.outputs.version_tag }}
+            type=semver,pattern=v{{major}}.{{minor}},value=${{ steps.tag.outputs.version_tag }}
+            type=semver,pattern=v{{major}},value=${{ steps.tag.outputs.version_tag }}
+
+      - name: login to GHCR
+        if: inputs.publish
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
```

Comment:
I think this represents the user who runs the workflow. How does it work for scheduled workflows? Do we need any setting to allow all code owners to publish packages?

Reply:
in our case this is always the GH bot user. This has to match the `password` value, which is the autogenerated token.